### PR TITLE
Add object detection

### PR DESCRIPTION
**Pull request by MicLi01**

### CODE REVIEW

File: src/cv/yolo_video_detect.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+def get_configs(args):
+    """ Retrieves the object detector model resources """
+    labelsPath = os.environ.get('YOLO_NAMES_PATH')
+    labels = open(labelsPath).read().strip().split("\n")
+
+    np.random.seed(int(time.time()))
+    colors = np.random.randint(0, 255, size=(len(labels), 3), dtype="uint8")
+
+    weightsPath = os.environ.get('YOLO_WEIGHTS_PATH')
+    configPath = os.environ.get('YOLO_CONFIG_PATH')
```
Comment (chc5) [inline src/cv/yolo_video_detect.py]: What happens if the environment variables doesn't exist in the OS or change all of a sudden in the OS? I suggest that you make a settings.py and make them as constants in that file and set defaults or errors if those variables don't exist.

File: src/cv/yolo_video_detect.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    try:
+        if imutils.is_cv2():
+            prop = cv2.cv.CV_CAP_PROP_FRAME_COUNT
+        else:
+            prop = cv2.CAP_PROP_FRAME_COUNT
+        total = int(vs.get(prop))
+        print("{} total frames in video".format(total))
+
+    except:
+        print("Could not determine # of frames in video")
```
Comment (chc5) [inline src/cv/yolo_video_detect.py]: Raise an error instead of printing.

File: src/cv/yolo_video_detect.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+def iterate_frames(args, vs, net, ln, colors, labels, total):
+    """ Iterate through each frame and pass it through the net """
+    writer = None
+    (W, H) = (None, None)
+    while True:
+        (grabbed, frame) = vs.read()
+        if writer is None:
+            fourcc = cv2.VideoWriter_fourcc(*"MJPG")
+            writer = cv2.VideoWriter(args["output"], fourcc, 30, (frame.shape[1], frame.shape[0]), True)
```
Comment (chc5) [inline src/cv/yolo_video_detect.py]: There should be 50 to 80 characters per line

File: src/cv/yolo_video_detect.py (hunk 1 above)
Comment (MicLi01) [reply src/cv/yolo_video_detect.py]: Added a settings file and added exceptions

File: src/cv/yolo_video_detect.py (hunk 2 above)
Comment (MicLi01) [reply src/cv/yolo_video_detect.py]: done

File: src/cv/yolo_video_detect.py (hunk 3 above)
Comment (MicLi01) [reply src/cv/yolo_video_detect.py]: modified

**Comment by chc5** [comment]
Make sure to make the commit message more specific and remove all the comments or put them in paragraph form.

**Pull request merged**